product/interfaces/graphql/product/dto/variantsToVariationSelectionsMapper.go

```diff
@@ -169,12 +169,10 @@ func (m *variantsToVariationSelectionsMapper) buildVariationSelectionOptions(att
 	var options []VariationSelectionOption
 
 	attributeGroup.eachAttributeInOrder(func(attribute *domain.Attribute) {
-		var option *VariationSelectionOption
+		option := m.createOptionWithoutActiveVariant(*attribute)
```

Comment:
@carstendietrich hmm, by removing this else branch you will always loop through all options twice when an active variant is set. The result is the same, but this is probably not what we want in terms of performance.

Reply:
I am aware of this, I guess it is more important to have the code easily understood in a few months than to have it micro-optimized. If there was an API call or a large calculation, I would stick to the else branch as well.

Reply:
Ok fine then